### ISSUE TITLE
Evaluate replacing frontpage React components with Flask/Jinja/vanilla JS components from ZenodoRDM

### ISSUE THREAD

**Issue by jennur** [inveniosoftware/invenio-app-rdm#2291]
In Zenodo RDM we have implemented the `CommunitiesCarousel` and `RecordsList` (recent uploads) component in pure Jinja and vanilla JS, for performance reasons. The question is if this should be moved to `invenio-app-rdm`/`invenio-communities` instead.

## About the ZenodoRDM components

Recent uploads component: 
- Identical to the React component
- Requires backend fetching in the frontend view function

Communities carousel:
- Currently only supports 1 item per page, otherwise identical
- Requires backend fetching in the frontend view function

**Comment by Samk13** [comment]
Indeed, In V12rc2 still the carousel animation is stuttering because the [current React component](https://github.com/inveniosoftware/invenio-communities/blob/68ee5d505f209cd7b5df900d4bf1d0b8ba3f2514/invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesCarousel/CarouselItem.js#L17) has unnecessary rerenders. It should be replaced with the [Carousel implementation](https://github.com/zenodo/zenodo-rdm/blob/04241b8b27918adea16c6c940cb4188a98151552/site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/communities-carousel.js#L113) from Zenodo.
